src/Cli/dotnet/ReleasePropertyProjectLocator.cs

```diff
@@ -0,0 +1,153 @@
+                    {
+                        shouldReturnNull = true;
+                        state.Stop(); // We don't want to override Configuration if ANY project in a sln uses a custom configuration
+                        return;
+                    }
+
+                    string configuration = projectData.GetPropertyValue(slnProjectConfigPropertytoCheck);
+                    if (!string.IsNullOrEmpty(configuration))
+                    {
+                        configuredProjects.Add(projectData); // we don't care about race conditions here
+                        configValues.Add(configuration);
+                    }
+                }
+            });
+
+            if (configuredProjects.Any() && configValues.Count > 1)
+            {
+                throw new GracefulException(CommonLocalizableStrings.TopLevelPublishConfigurationMismatchError);
+            }
+
+            return shouldReturnNull || configuredProjects.Count == 0 ? null : configuredProjects.First();
+        }
+
+        /// <summary>
+        /// Provide a CLI input to change configuration based on 
+        /// a boolean that may or may not exist in the targeted project.
+        /// <param name="defaultedConfigurationProperty">The boolean property to check the project for. Ex: PublishRelease</param>
+        /// <param name="slnOrProjectArgs">The arguments or solution passed to a dotnet invocation.</param>
+        /// <param name="configOption">The arguments passed to a dotnet invocation related to Configuration.</param>
+        /// </summary>
+        /// <returns>Returns a string such as -property:configuration=value for a projects desired config. May be empty string.</returns>
+        public IEnumerable<string> GetCustomDefaultConfigurationValueIfSpecified(
+            ParseResult parseResult,
+            string defaultedConfigurationProperty,
+            IEnumerable<string> slnOrProjectArgs,
+            Option<string> configOption
+            )
+        {
+            ProjectInstance project = null;
+
+            IEnumerable<string> calledArguments = parseResult.Tokens.Select(x => x.ToString());
+            IEnumerable<string> slnProjectAndCommandArgs = slnOrProjectArgs.Concat(calledArguments);
+            project = GetTargetedProject(slnProjectAndCommandArgs, defaultedConfigurationProperty);
+
+            if (project != null)
+            {
+                string releaseMode = "";
+                string releasePropertyFlag = project.GetPropertyValue(defaultedConfigurationProperty);
+                if (!string.IsNullOrEmpty(releasePropertyFlag))
+                    releaseMode = releasePropertyFlag == "true" ? "Release" : "Debug";
+
+                if (!ConfigurationAlreadySpecified(parseResult, project, configOption) && !string.IsNullOrEmpty(releaseMode))
+                    return new List<string> { $"-property:configuration={releaseMode}" };
+            }
+            return Array.Empty<string>();
```

Comment:
Nit: I think this is slightly better:

```suggestion
            return Enumerable.Empty<string>();
```